README.md

```diff
@@ -52,6 +57,23 @@ For information about the API, see the [API documentation](API.md).
 
 For information about testing the service, see the [Testing documentation](TESTING.md).
 
+## Secrets
+
+Secrets for this project follow the naming convention `${EnvironmentType}/${DNSRecord}/honeybadger_api_key`. Our security model controls access to secrets based on the prefix, and will deny read access to any secret not prefixed with `development/`, so we tell our application's Cloudformation template whether we are in a development, test, or production environment type via [config files](cicd/3-app/aiproxy/config).
```

Comment:
Is this a new naming convention (with the hostname in the name) we're introducing? Just wondering because I haven't see it in the current secrets in prod.

Also, are we using wildcard secrets in the actual Role policy with this naming, or are we linking ARNs directly to roles? Direct ARNs might be more secure, but I get if it's not as convenient.

Reply:
Yes, this is declaring that the 'name' of the environment is the DNS name. We've had some longstanding issues with the use of the key `environment` in the code-dot-org repo. When we run tests locally, the application is running with `environment=test` and when we deploy adhocs, every adhoc has `environment=adhoc`, so there's some inconsistency with what it would mean to say "The environment is X". We have a scenario in Javabuilder where we actually have two 'production' environments.

The solution we've been headed towards is to distinguish between 'environment' which is is the unique name of the set of resources, and 'environment type' which defines the features and configuration of the environment.

For the second question, yes, [we're using wildcards](https://github.com/code-dot-org/code-dot-org/blob/staging/aws/cloudformation/iam.yml.erb#L204C22-L204C22) to define what secrets engineers can access.

Reply:
I was wondering about the ECS execution role.  This is the one for the existing fargate, I think:


To allow this to get the secret, we would need to modify that policy to be able to access the secret, something like:

```
 "Statement": [
        {
            "Effect": "Allow",
            "Action": "secretsmanager:GetSecretValue",
            "Resource": "arn:aws:secretsmanager:us-east-1:475661607190:secret:production/aiproxy.code.org/honeybadger_api_key"
        }
    ]
```

I just was wondering if we would instead to a wildcard, allowing this ECS to access all production secrets with wildcards like:
- arn:aws:secretsmanager:us-east-1:475661607190:secret:production/aiproxy.code.org/*

OR

- arn:aws:secretsmanager:us-east-1:475661607190:secret:production/*